Gems/Terrain/Code/Source/Components/TerrainWorldRendererComponent.cpp

```diff
@@ -0,0 +1,225 @@
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZStd::vector<AZ::Crc32>({ AZ_CRC_CE("Level") }))
+                    ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true);
+            }
+        }
+    }
+
+    void TerrainWorldRendererComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& services)
+    {
+        services.push_back(AZ_CRC_CE("TerrainRendererService"));
+    }
+
+    void TerrainWorldRendererComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& services)
+    {
+        services.push_back(AZ_CRC_CE("TerrainRendererService"));
+    }
+
+    void TerrainWorldRendererComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& services)
+    {
+        services.push_back(AZ_CRC_CE("TerrainService"));
+    }
+
+    void TerrainWorldRendererComponent::Reflect(AZ::ReflectContext* context)
+    {
+        TerrainWorldRendererConfig::Reflect(context);
+
+        AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serialize)
+        {
+            serialize->Class<TerrainWorldRendererComponent, AZ::Component>()->Version(0)->Field(
+                "Configuration", &TerrainWorldRendererComponent::m_configuration);
+        }
+    }
+
+    TerrainWorldRendererComponent::TerrainWorldRendererComponent(const TerrainWorldRendererConfig& configuration)
+        : m_configuration(configuration)
+    {
+    }
+
+    TerrainWorldRendererComponent::~TerrainWorldRendererComponent()
+    {
+        if (m_terrainRendererActive)
+        {
+            Deactivate();
+        }
+    }
+
+    void TerrainWorldRendererComponent::Activate()
+    {
+        // On component activation, register the terrain feature processor with Atom and the default scene.
+
+        AZ::RPI::FeatureProcessorFactory::Get()->RegisterFeatureProcessor<Terrain::TerrainFeatureProcessor>();
+        if (auto rpi = AZ::RPI::RPISystemInterface::Get(); rpi)
+        {
+            if (auto defaultScene = rpi->GetDefaultScene(); defaultScene)
+            {
+                AZ::RPI::Scene* scene = defaultScene.get();
+                scene->EnableFeatureProcessor<Terrain::TerrainFeatureProcessor>();
+            }
+        }
+
+        AzFramework::Terrain::TerrainDataNotificationBus::Handler::BusConnect();
+        m_terrainRendererActive = true;
+    }
+
+    void TerrainWorldRendererComponent::Deactivate()
+    {
+        // On component deactivation, unregister the feature processor and remove it from the default scene.
+
+        m_terrainRendererActive = false;
+        AzFramework::Terrain::TerrainDataNotificationBus::Handler::BusDisconnect();
+
+        if (auto rpi = AZ::RPI::RPISystemInterface::Get(); rpi)
+        {
+            if (auto defaultScene = rpi->GetDefaultScene(); defaultScene)
```

Comment:
You should be able to call `RPI::Scene::GetSceneForEntityContextId()` to get the correct scene for the EntityContext that this component is in. Getting the default scene is usually the wrong thing to do.
```
// Find the entity context for the entity ID.
AzFramework::EntityContextId entityContextId = AzFramework::EntityContextId::CreateNull();
AzFramework::EntityIdContextQueryBus::EventResult(entityContextId, entityId, &AzFramework::EntityIdContextQueryBus::Events::GetOwningContextId);
```
Even better would be to add a function to RPI::Scene that allows you to get a Scene from an EntityId and handles the entity context lookup for you.

Reply:
Fixed!  Switched to the better scene lookup, but stopping short of creating the helper function.  I don't want to expand the scope of my PR.